File: programming-challenges.com/The3n+1problem/c/a.c
/* Copyright (C) 2016 by iamslash */

#include <stdio.h>
#define max(a,b)
  ({ __typeof__ (a) _a = (a);
    __typeof__ (b) _b = (b);
    _a > _b ? _a : _b; })

int CACHE[100000];

int cycle_len(int n) {
  return n;
}

int main() {
  int I, J;

  while (scanf("%d %d", &I, &J) != EOF) {
    int max_cycle_len = 0;
    for (int i = I; i <= J; ++i) {
      max_cycle_len = max(max_cycle_len, cycle_len(i));
    }
    printf("%d %d %d\n", I, J, max_cycle_len);
  }

  return 0;
}

File: programming-challenges.com/The3n+1problem/a.py
# -*- coding: utf-8 -*-
# Copyright (C) 2016 by iamslash
# > python3 a.py

import sys

CACHE = []
I = 0
J = 0

def cycle_len(n):
    
    
    return 0

if __name__ == "__main__":
    for line in sys.stdin:
        CACHE = [-1] * 1000000
        
        line = line.strip()
        l = line.split(' ')
        I = int(l[0])
        J = int(l[1])

        r = 0
        for n in range(I, J+1):
            r = max(r, cycle_len(n))
            
        print('{} {} {}'.format(I, J, r))
